src/modules/ekf2/EKF/control.cpp

```diff
@@ -845,9 +845,16 @@ void Ekf::controlHeightSensorTimeouts()
 	// check if height is continuously failing because of accel errors
 	const bool continuous_bad_accel_hgt = isTimedOut(_time_good_vert_accel, (uint64_t)_params.bad_acc_reset_delay_us);
 
-	// check if height has been inertial deadreckoning for too long
-	// in vision hgt mode check for vision data
-	const bool hgt_fusion_timeout = isTimedOut(_time_last_hgt_fuse, (uint64_t)5e6);
+	// check if height has been inertial deadreckoning for too long depending on the height source
+	bool hgt_fusion_timeout = false;
+
+	if(_control_status.flags.baro_hgt) {
+		hgt_fusion_timeout = isTimedOut(_time_last_hgt_fuse, (uint64_t)5e6);
+
+	} else {
+		hgt_fusion_timeout = isTimedOut(_time_last_hgt_fuse, (uint64_t)3e5);
```

Comment:
What as the reasoning behind treating all sources of height other than Baro such as GPS as requiring a smaller timeout? I would have thought that the main reason for using a smaller timeout value is when we we do not have an independent source of velocity measurement.

Reply:
I agree, the main reason is to keep the delay before falling back as small as possible.